alivee/place_search_engine/utils.py

```diff
@@ -0,0 +1,139 @@
+# Put Standard Library Imports Here:
+import math
+import time
+
+# Put Third Party/Django Imports Here:
+from geopy.distance import vincenty
+
+# Put Alivee Imports Here:
+from general_utils.requests.google.place_search.constants import PlaceSearchPlaceType
+from general_utils.requests.google.place_search.exceptions import ExceedMaximumMainRequestRetryCountException
+from general_utils.requests.google.place_search.utils import PlaceSearch
+from place_search_engine.constants import CityBoundaryPointsInfo
+
+MAX_BACKFILL_RADIUS_IN_METERS = 1000
+
+
+def _getDistanceBetweenTwoPoints(point1, point2):
+	'''
+	Private helper method to get distance in meters for 2 points.
+
+	param point1: First point in tuple format: (lat, lng).
+	param point2: Second point in tuple format: (lat, lng).
+	return: Distance of 2 points in meters.
+	'''
+	return vincenty(point1, point2).meters
+
+
+def _getBackfillCenterPoints(upperLeftPoint, lowerRightPoint, width, height):
+	'''
+	Private helper method to get backfill center points for rectangle areas.
+
+	param width: The real distance between left and right points, in meters.
+	param height: The real distance between upper and lower points, in meters. 
+	return: A tuple (backfillRadius, backfillPoints)
+	'''
+	# Potential count of points in width and height direction when we use max radius to divide. Call it 
+	# `potential` since not all points will be included according to our `checkerboard` style backfill strategy.
+	potentialPointsCountInWidthDir = int(math.ceil(width / MAX_BACKFILL_RADIUS_IN_METERS)) + 1
+	potentialPointsCountInHeightDir = int(math.ceil(height / MAX_BACKFILL_RADIUS_IN_METERS)) + 1
+
+	backfillCenterPointsToInclude = []
+	(firstLat, firstLng), (lastLat, lastLng) = upperLeftPoint, lowerRightPoint
+	isPointToIncludeInHeightDir = True
+
+	currentLat, currentLng = firstLat, firstLng
+	for heightIndex in xrange(potentialPointsCountInHeightDir):
+		# Set `currentLat` to current value in height direction by using height index.
+		currentLat = currentLat + (lastLat - currentLat) * heightIndex / (potentialPointsCountInHeightDir - 1)
+		isPointToIncludeInWidthDir = isPointToIncludeInHeightDir
+
+		currentLng = firstLng
+		for widthIndex in xrange(potentialPointsCountInWidthDir):
+			# Set `currentLng` to current value in width direction by using width index.
+			currentLng = currentLng + (lastLng - currentLng) * widthIndex / (potentialPointsCountInWidthDir - 1)
+
+			if isPointToIncludeInWidthDir:
+				backfillCenterPointsToInclude.append([currentLat, currentLng])
+
+			isPointToIncludeInWidthDir = not isPointToIncludeInWidthDir
+
+		isPointToIncludeInHeightDir = not isPointToIncludeInHeightDir
+
+	backfillRadius = max(width / ((potentialPointsCountInWidthDir - 1)), height / (potentialPointsCountInHeightDir - 1))
+
+	return backfillRadius, backfillCenterPointsToInclude
+
+
+def searchPlacesByTypeInArea(key, place_type=PlaceSearchPlaceType.RESTAURANT.value, boundaryPointsInfo=None, 
```

Comment:
Actually a WIP util method, can be the candidate of our backfill method.
Don't dig deep into this method, just use for testing search coverage for area for now.